sql/driver/wire.go

```diff
@@ -80,14 +100,17 @@ func (d Datum) Value() (driver.Value, error) {
 		val = t.BytesVal
 	case *Datum_StringVal:
 		val = t.StringVal
+	case *Datum_DateVal:
+		val = Date{t.DateVal.GoTime()}
```

Comment:
I added a commit that replaces `Time.Truncate` with a zone-aware implementation - I'm pretty certain at this point that Timestamp and Date are consistent, but could you have another look?